devices/bmp180/bmp180.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package bmp180 controls a Bosch BMP180 device over I²C.
+//
+// Datasheet
+//
+// https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+package bmp180
+
+import (
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/devices"
+)
+
+// Oversampling affects how much time is taken to measure pressure.
+type Oversampling uint8
+
+// Possible oversampling values.
+const (
+	No  Oversampling = 0
+	O2x Oversampling = 1
+	O4x Oversampling = 2
+	O8x Oversampling = 3
+)
+
+// Dev is an handle to a bmp180.
+type Dev struct {
+	d  conn.Conn
+	c  calibration
+	os Oversampling
+}
+
+func (d *Dev) String() string {
+	return fmt.Sprintf("BMP180{%s}", d.d)
+}
+
+const (
+	ctrlCommandTemperature = 0x0E
+	ctrlCommandPressure    = 0x14
+)
+
+// bit offsets in regCtrlMeas
+const (
+	ctrlMeasurementControl = 0
+	ctrlStartConversion    = 5
+	ctrlOversampling       = 6
+)
+
+// Sense returns measurements as °C and kPa.
+func (d *Dev) Sense(env *devices.Environment) error {
+	// start conversion for temperature
+	var cmd byte
+	cmd = 1 << ctrlStartConversion
+	cmd |= ctrlCommandTemperature << ctrlMeasurementControl
+
+	if err := d.d.Tx([]byte{regCtrlMeas, cmd}, nil); err != nil {
+		return err
+	}
+
+	time.Sleep(100 * time.Millisecond)
+
+	// read value
+	var tempBuf [2]byte
+	if err := d.readReg(regOutMSB, tempBuf[:]); err != nil {
+		return err
+	}
+
+	UT := uint16(tempBuf[0])<<8 | uint16(tempBuf[1])
```

Comment:
You are right there is no meaning inside a function, but the code style is to always use lower case names for locally scoped symbols.